### PR TITLE
AG

### PR DESCRIPTION
**Pull request by Alyelfekyy**

### CODE REVIEW

File: routes/api/users.js [hunk 1]
```diff
@@ -61,4 +61,18 @@ router.put("/:username",(request,response) =>{
      response.send({msg:"User deleted",data:newdata});
 })
 
+router.put("/:username/grades",(request,response) =>{
+    const x=request.body;
```
Comment (YahiaBadr) [inline routes/api/users.js]: rename x to be understandable by the reader.
Like `const body = request.body` or another better option,
`const courseName = request.body.courseName` and make the other constants for `grade` and `creditHours`.

File: routes/api/users.js [hunk 2]
```diff
@@ -61,4 +61,18 @@ router.put("/:username",(request,response) =>{
...
+    for(let i =0;i< users.length;i++){
+        if (users[i].username===request.params.username){
+            users[i].grades.push({"courseName":x.courseName, "grade":x.grade, "creditHours":x.creditHours})
+            break;
+           
+        }
+        
+    }
+    console.log(users)
```
Comment (YahiaBadr) [inline routes/api/users.js]: remove the console.log. Leave the code clean.

File: routes/api/users.js [hunk 3]
```diff
@@ -61,4 +61,18 @@ router.put("/:username",(request,response) =>{
      response.send({msg:"User deleted",data:newdata});
 })
 
+router.put("/:username/grades",(request,response) =>{
```
Comment (YahiaBadr) [inline routes/api/users.js]: the route must be clean and descriptive,
`/addGrades/:username`,
always leave the params at the end.

**Review by YahiaBadr — CHANGES_REQUESTED** [review]

File: routes/api/users.js [hunk 4]
```diff
@@ -61,18 +61,20 @@ router.put("/:username",(request,response) =>{
      response.send({msg:"User deleted",data:newdata});
 })
 
-router.put("/:username/grades",(request,response) =>{
-    const x=request.body;
+router.put("/grades/:username",(request,response) =>{
```
Comment (YahiaBadr) [inline routes/api/users.js]: the route mus be descriptive `/addGrade/:username`

File: routes/api/users.js [hunk 5]
```diff
@@ -61,18 +61,20 @@ router.put("/:username",(request,response) =>{
...
+    const coursename=request.body.courseName;
```
Comment (YahiaBadr) [inline routes/api/users.js]: camel case `courseName`

File: routes/api/users.js [hunk 6]
```diff
@@ -61,18 +61,20 @@ router.put("/:username",(request,response) =>{
...
+    const thegrade=request.body.grade;
```
Comment (YahiaBadr) [inline routes/api/users.js]: the variable name must be the same as the one in the body to be understandable throw the code.
Change `thegrade` to be `grade`

File: routes/api/users.js [hunk 7]
```diff
@@ -61,18 +61,20 @@ router.put("/:username",(request,response) =>{
...
+    const credithours=request.body.creditHours
```
Comment (YahiaBadr) [inline routes/api/users.js]: camel case error, `creditHours`.

**Review by YahiaBadr — CHANGES_REQUESTED** [review]

File: routes/api/users.js (hunk 5 above)
Comment (Alyelfekyy) [reply routes/api/users.js]: msh fahem ya3ny a3ml eh akhleha ezay?:D

Comment (YahiaBadr) [reply routes/api/users.js]: `const courseName`

File: routes/api/users.js (hunk 4 above)
Comment (YahiaBadr) [reply routes/api/users.js]: @alynamlaa

File: routes/api/users.js [hunk 8]
```diff
@@ -61,4 +61,20 @@ router.put("/:username",(request,response) =>{
      response.send({msg:"User deleted",data:newdata});
 })
 
+router.put("/addGrades/:username",(request,response) =>{
+    const courseName=request.body.courseName;
+    const grade=request.body.grade;
+    const creditHours=request.body.creditHours
+    for(let i =0;i< users.length;i++){
+        if (users[i].username===request.params.username){
+            users[i].grades.push({"courseName":courseName, "grade":grade, "creditHours":creditHours})
```
Comment (YahiaBadr) [inline routes/api/users.js]: you don't have to write the field between quotes.

File: routes/api/users.js [hunk 9]
```diff
@@ -61,4 +61,20 @@ router.put("/:username",(request,response) =>{
      response.send({msg:"User deleted",data:newdata});
 })
 
+router.put("/addGrades/:username",(request,response) =>{
```
Comment (YahiaBadr) [inline routes/api/users.js]: addGrade!!!! NO (s)

**Review by YahiaBadr — APPROVED** [review]

**Pull request merged**